addons/account/models/chart_template.py

```diff
@@ -972,6 +992,41 @@ def _parse_csv(self, template_code, model, module=None):
                 _logger.debug("No file %s found for template '%s'", model, module)
         return res
 
+    def _get_template_model_records_with_untranslated_translatable_name(self, langs, companies):
+        if not langs or not companies:
+            return []
+
+        queried_models = [model for model in TEMPLATE_MODELS if self.env[model]._fields['name'].translate]
+
+        for model in queried_models:
+            self.env[model].flush_model(['id', 'company_id', 'name'])
+        self.env['ir.model.data'].flush_model(['res_id', 'model', 'name'])
+
+        WHERE_clause_missing_translation = " OR ".join([
+            f"model.name->>'{lang}' IS NULL"
+            for lang in langs
+        ])
+
+        queries = [
+            f'''
+             SELECT '{model}' AS model,
+                    model_data.name AS xmlid,
+                    model.name AS name,
+                    model_data.module AS module
+             FROM {model_table} model
+             JOIN ir_model_data model_data
+                 ON  model_data.model = '{model}'
+                 AND model.id = model_data.res_id
+             WHERE {WHERE_clause_missing_translation}
+                 AND model.name->>'en_US' IS NOT NULL
+                 AND model.company_id IN ({','.join([str(id) for id in companies.ids])})
+            '''
+            for model_table, model in [(self.env[model]._table, model) for model in queried_models]
+        ]
```

Comment:
```suggestion
        queries = [f"""
             SELECT '{model}' AS model,
                    model_data.name AS xmlid,
                    model.name AS name,
                    model_data.module AS module
               FROM {self.env[model]._table} model
               JOIN ir_model_data model_data ON model_data.model = '{model}'
                                            AND model.id = model_data.res_id
              WHERE {where_clause_missing_translation}
                AND model.name->>'en_US' IS NOT NULL
                AND model.company_id IN %(company_ids)s
        """ for model in queried_models]
```

Reply:
It would also be better if `'{model}'` was properly escaped too, soon a lot easier in master

Reply:
Thanks for the update. I put `model` and `lang` (too) into the `params` in the new version.